primitives/src/v0.rs

```diff
@@ -114,7 +114,8 @@ impl MallocSizeOf for ValidatorId {
 }
 
 /// Index of the validator is used as a lightweight replacement of the `ValidatorId` when appropriate.
-pub type ValidatorIndex = u32;
+#[derive(Debug, Eq, Ord, PartialEq, PartialOrd, Hash, Copy, Clone, Encode, Decode, MallocSizeOf)]
```

Comment:
some traits like `MallocSizeOf` should be gated by `std` feature

Reply:
As should `Debug`

Reply:
Thank you guys! Will fix in a minute!

Reply:
Also, we might want to specify #[repr(transparent)]` to guarantee backwards compatibility, but not 100% sure it's actually needed

https://doc.rust-lang.org/nomicon/other-reprs.html#reprtransparent

Reply:
Backwards compatibility with what? Are we using this struct in an unsafe context?

Reply:
We don't have a FFI anywhere.

Reply:
Yeah, I guess scale-encoding should be the same, so nvm.

Reply:
Anyhow, I was wondering about this as well. So glad it got discussed :-)